### PR TITLE
Optimize transactions query

### PR DESCRIPTION
**Pull request by xin-hedera**
**Detailed description**:

- Move json aggregation of transfers into subquery so can put them as select fields
 
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
Previously, for a transaction of N crypto transfers, the query returns N rows. With the optimization, there will be exactly one row in the response for one transaction, with every type of transfers now aggregated in a json string.

Test show for the default 500 limit, the amount of data sent from pg server to client for a `/api/v1/transactions` query is reduced from 822kb to 293kb.

Benchmark in perfnet shows 10 nodejs processes can do 53 QPS v.s. 40.

**Checklist**

- [ ] Documentation added
- [x] Tests updated

### CODE REVIEW

**Review by steven-sheehy — APPROVED** [review]
LGTM

File: hedera-mirror-rest/transactions.js [hunk 1]
```diff
@@ -27,55 +27,94 @@ const TransactionId = require('./transactionId');
...
-       t.scheduled,
-       t.entity_id,
-       t.transaction_bytes`;
+    ${fields.join(',\n')}
+  `;
+};
+
+/**
+ * Creates crypto transfer list from aggregated array of JSON objects in the query result. Note if the
+ * cryptoTransferList is undefined, any empty array is returned.
```
Comment (ijungmann) [inline hedera-mirror-rest/transactions.js:95]: nit:
```suggestion
 * cryptoTransferList is undefined, an empty array is returned.
```

**Review by ijungmann — APPROVED** [review]
Looks good, one tiny nit you may wish to address.

**Review by Nana-EC — DISMISSED** [review]

File: hedera-mirror-rest/transactions.js (hunk 1 above)
Comment (xin-hedera) [reply hedera-mirror-rest/transactions.js]: fixed

**Review by Nana-EC — APPROVED** [review]

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**